### PR TITLE
Send avatar info in game lobby

### PR DESCRIPTION
**Pull request by GodFuper**
Sending player avatar info (name/tooltip) in game lobby for displaying near nickname.

last #943 #942

### CODE REVIEW

File: src/main/java/com/faforever/client/fa/LaunchCommandBuilder.java [hunk 1]
```diff
@@ -169,6 +182,17 @@ public LaunchCommandBuilder rehost(boolean rehost) {
       command.add(localIp + ":" + localGpgPort);
     }
 
+
+    if(avatarUrl != null) {
+	  String urlTOfilename = null; // In lobby it is easier to use the name of the avatar file than the link itself.
```
Comment (Geosearchef) [inline src/main/java/com/faforever/client/fa/LaunchCommandBuilder.java]: formatting off

File: src/main/java/com/faforever/client/fa/LaunchCommandBuilder.java [hunk 2]
```diff
@@ -169,6 +182,17 @@ public LaunchCommandBuilder rehost(boolean rehost) {
...
+	  urlTOfilename = Paths.get(URI.create(avatarUrl).getPath()).getFileName().toString();
```
Comment (Geosearchef) [inline src/main/java/com/faforever/client/fa/LaunchCommandBuilder.java]: why do you first create and then assign inside the same block?
(multiple times)

File: src/main/java/com/faforever/client/fa/LaunchCommandBuilder.java [hunk 3]
```diff
@@ -169,6 +182,17 @@ public LaunchCommandBuilder rehost(boolean rehost) {
...
+      command.add("/avatarurl");   // If you change the command, you need to remember to change in lobby.
+      command.add(String.valueOf(urlTOfilename));
```
Comment (Geosearchef) [inline src/main/java/com/faforever/client/fa/LaunchCommandBuilder.java]: remove `String.valueOf`, urlTOfilename is already a string

File: src/main/java/com/faforever/client/fa/LaunchCommandBuilder.java [hunk 4]
```diff
@@ -169,6 +182,17 @@ public LaunchCommandBuilder rehost(boolean rehost) {
       command.add(localIp + ":" + localGpgPort);
```
Comment (Geosearchef) [inline src/main/java/com/faforever/client/fa/LaunchCommandBuilder.java]: Not part of your changes, but gonna mention it anyways ;P  
`String.format("%s:%d", localIp, localGpgPort)`

File: src/main/java/com/faforever/client/fa/LaunchCommandBuilder.java (hunk 1 above)
Comment (Geosearchef) [reply src/main/java/com/faforever/client/fa/LaunchCommandBuilder.java]: also, java convention would be to name this variable `urlToFilename`

**Review by Geosearchef — CHANGES_REQUESTED** [review]
PR looks good, but:

This lead to a long discussion in #game-developement with the result that player's avatar, rating and games need to be checked by each client, either by reading a file/contacting API/server /gpgnet.  Therefore the avatar will not be send by other players which hand it to their client as a command line parameter.

File: src/main/java/com/faforever/client/fa/LaunchCommandBuilder.java (hunk 2 above)
Comment (GodFuper) [reply src/main/java/com/faforever/client/fa/LaunchCommandBuilder.java]: @Geosearchef Like I tried to do so, but the Travis (plugin in github) gave an error.

**Comment by Geosearchef** [comment]
I also want to mention that you don't need to recreate your pull request unless you change the branch. You can just force push to your current branch and the PR will update automatically. So if you wanna e.g. implement requested changes you can just commit them while ticking the `amend` box during the process and the force push. Github will automatically detect when something I requested a change on changed and marked the request as processed.

File: src/main/java/com/faforever/client/fa/LaunchCommandBuilder.java (hunk 2 above)
Comment (Geosearchef) [reply src/main/java/com/faforever/client/fa/LaunchCommandBuilder.java]: Pls do it again and let me take a look. I can't believe that, maybe there was something else going wrong. If it really happened due to this we need to fix travis, not the code ;).

**Comment by GodFuper** [comment]
@Geosearchef 
Your long discussions in #game-development are at the stage of conversations. "need to be checked by each client, either by reading a file/contacting API/server /gpgnet." The implementation of this may take months or even more. That before, that now not was protection from this. Currently there is no API/server connection. So this is no reason to cancel this development. This is the minimum that will make avatars more attractive. Although I also believe that it is necessary to communicate with the API and the server.

**Comment by 1-alex98** [comment]
There seem to be user that actively exploit such security breaches. Eventhough that is really stupid. So we need to do it securely

**Comment by 1-alex98** [comment]
Should we every change to the java server we will be able to validate users avatars

**Comment by GodFuper** [comment]
Axel, well, have you already done the modification for validate players?

**Comment by 1-alex98** [comment]
downlord did it, not me I reviewed it... but as i said on java-server

**Comment by GodFuper** [comment]
so now let's insert the avatar in the lobby?

**Comment by 1-alex98** [comment]
java-server is not deployed and we would also need to add this check once it is deployed

**Comment by GodFuper** [comment]
Well, if my help will be need, write me.

**Comment by 1-alex98** [comment]
You can help where you want, there are lots of issues open across all different repos. But deciding to deploy java-server can not be helped I assume. You could help improving it if you wanted

**Comment by 1-alex98** [comment]
We will close this. We can reopen if server support for this is up.

**Pull request closed without merging**